File: frontend/src/responses/incident.response.ts
import {ProfileResponse} from "./profile.response.ts";
import {IncidentType} from "../enums/incident-type.enum.ts";
import {IncidentSeverity} from "../enums/incident-severity.enum.ts";
import {IncidentStatus} from "../enums/incident-status.enum.ts";
import {IncidentCheckResponse} from "./incident-check.response.ts";

export interface IncidentResponse {
    id: string,
    title: string,
    system: string,
    time: string,
    status: IncidentStatus,
    severity: IncidentSeverity,
    type: IncidentType,
    checks: IncidentCheckResponse[],
    creator: ProfileResponse,
    assignee: ProfileResponse,
    discussionId: string,
    created: string,
    updated: string
}

File: backend/src/main/java/ch/oliumbi/gyroscope/endpoints/incident/responses/IncidentResponse.java
package ch.oliumbi.gyroscope.endpoints.incident.responses;

import ch.oliumbi.gyroscope.endpoints.discussion.responses.DiscussionResponse;
import ch.oliumbi.gyroscope.endpoints.profile.responses.ProfileResponse;
import ch.oliumbi.gyroscope.core.incident.enums.IncidentSeverity;
import ch.oliumbi.gyroscope.core.incident.enums.IncidentStatus;
import ch.oliumbi.gyroscope.core.incident.enums.IncidentType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class IncidentResponse {
    private UUID id;
    private String title;
    private String system;
    private LocalDateTime time;
    private IncidentStatus status;
    private IncidentSeverity severity;
    private IncidentType type;
    private List<IncidentCheckResponse> checks;
    private ProfileResponse creator;
    private ProfileResponse assignee;
    private UUID discussionId;
    private LocalDateTime created;
    private LocalDateTime updated;
}
